### PR TITLE
[release-4.3] Bug 1803762: Revendor MAO with new node draining feature check existence before node deletion

### PR DESCRIPTION
**Pull request by alexander-demichev**
Revendor MAO with a new node draining feature and check for instance existence before proceeding with deletion.
https://github.com/openshift/machine-api-operator/pull/464 
https://github.com/openshift/machine-api-operator/pull/490
https://github.com/openshift/machine-api-operator/pull/495

### CODE REVIEW

**Comment by alexander-demichev** [comment]
We should wait before https://github.com/openshift/machine-api-operator/pull/495 gets merged

File: Makefile [hunk 1]
```diff
@@ -15,6 +15,13 @@
 # If you update this file, please follow
 # https://suva.sh/posts/well-documented-makefiles
 
+GO111MODULE = on
+export GO111MODULE
+GOFLAGS += -mod=vendor
```
Comment (JoelSpeed) [inline Makefile:20]: This can cause a double add of the mod flag which causes issue
```suggestion
GOFLAGS ?= -mod=vendor
```

File: Makefile [hunk 2]
```diff
@@ -268,12 +84,12 @@ fmt:
 	go fmt ./pkg/... ./cmd/...
 
 .PHONY: goimports
-goimports: ## Go fmt your code
+    goimports: ## Go fmt your code
```
Comment (JoelSpeed) [inline Makefile:87]: I think this tab is not right?

File: cmd/manager/main.go [hunk 3]
```diff
@@ -19,12 +19,12 @@ package main
 import (
 	"flag"
 
-	clusterapis "github.com/openshift/cluster-api/pkg/apis"
-	"github.com/openshift/cluster-api/pkg/client/clientset_generated/clientset"
-	capimachine "github.com/openshift/cluster-api/pkg/controller/machine"
+	"github.com/openshift/machine-api-operator/pkg/apis/machine/v1beta1"
```
Comment (JoelSpeed) [inline cmd/manager/main.go:22]: Nit: we'd normally import this as `machinev1`

File: hack/e2e.sh [hunk 4]
```diff
@@ -2,11 +2,9 @@
...
+unset GOFLAGS
+tmp="$(mktemp -d)"
 
-ACTUATOR_PKG="github.com/openshift/cluster-api-actuator-pkg"
+git clone "https://github.com/openshift/cluster-api-actuator-pkg.git" "$tmp"
 
-go get -u -d "${ACTUATOR_PKG}/..."
-
-exec make --directory="${GOPATH}/src/${ACTUATOR_PKG}" test-e2e
+exec make -C "$tmp" test-e2e
```
Comment (JoelSpeed) [inline hack/e2e.sh:10]: Nit, file should end with a new line

**Comment by JoelSpeed** [comment]
As this is just a cherry-pick, we can ignore my comments, they seem to not be breaking anything

/lgtm

**Comment by alexander-demichev** [comment]
@sdodson Can I get `cherry-pick-approved` ?

**Comment by enxebre** [comment]
/hold
@alexander-demichev can you please validate why `e2e-azure-operator` is consistently failing?

**Comment by michaelgugino** [comment]
/hold

This is a massive change, and much more is changing in this PR than just the described bug.

Since this code is in 4.4, and has shipped, I don't think we should merge this.  Just close/wontfix.

**Comment by sdodson** [comment]
/close
The bug was CLOSED WONTFIX so closing out this PR as well.

**Pull request closed without merging**